src/pages/index.js

```diff
@@ -19,39 +21,82 @@ import {
   profileTitle,
   profileDescription,
   cardListEl,
+  cardTemplate,
   previewImagepopupWindow,
   previewImageElement,
   previewImageTitle,
   profileEditForm,
   addNewCardForm,
-  cardTemplate,
+  avatarEditpencil,
+  userAvatarImage,
 } from "../utils/constants.js";
 // ! ||--------------------------------------------------------------------------------||
 // ! ||                                   FUNCTIONS;                                   ||
 // ! ||--------------------------------------------------------------------------------||
 
 //RENDER CARD FUNCTION
 
-function createCard(item) {
-  const card = new Card(item, cardTemplate, handlePreviewPicture);
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    cardTemplate,
+    handlePreviewPicture,
+    handleDeleteCardClick,
+    handleLikeButtonClick
+  );
   return card.getView();
 }
 // ! ||--------------------------------------------------------------------------------||
 // ! ||                                 EVENT HANDLERS                                 ||
 // ! ||--------------------------------------------------------------------------------||
 
+// EDIT AVATAR HANDLER
+
+function handleUpdateAvatarSubmit(input) {
+  updateAvatarPopupForm.setLoading(true);
+  api
+    .updateAvatar(input.link)
+    .then((res) => {
+      userInfo.setAvatar(res.avatar);
+      updateAvatarPopupForm.close();
+    })
+    .catch((err) => {
+      alert(`Error! ${err}`);
+    })
+    .finally(() => editProfilePopupForm.setLoading(false));
+}
+
 // PROFILE EDIT HANDLER
 
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data);
-  editProfilePopupForm.close();
+  editProfilePopupForm.setLoading(true);
+  api
+    .updateUserInfo(data)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+      editProfilePopupForm.close();
+    })
+    .catch((err) => {
+      alert(`Error! ${err}`);
+    })
+    .finally(() => editProfilePopupForm.setLoading(false));
 }
 
 // ADD NEW CARD HANDLER
 
 function handleAddNewCardSubmit(data) {
-  sectionCard.addItem(createCard({ name: data.title, link: data.link }));
-  addNewCardPopupForm.close();
+  addNewCardPopupForm.setLoading(true);
+  api
+    .addCard(data)
+    .then((card) => {
+      const newCard = createCard(card);
+      sectionCard.addItem(newCard);
+      addNewCardPopupForm.close();
+    })
+    .catch((err) => {
+      alert(`Error! ${err}`);
+    })
+    .finally(() => editProfilePopupForm.setLoading(false));
```

Comment:
`editProfilePopupForm.setLoading(false)` is only for the profile popup but you need to use `addNewCardPopupForm`